feat(pydantic): Support use_pydantic_alias parameter for pydantic conversion (#1546)
strawberry-graphql/strawberry: RELEASE.md

Release type: patch

> Adds support for use_pydantic_alias parameter in pydantic model conversion.
> Decides if the graphql field name should use the alias name or not.

Reviewer: ```suggestion
Adds support for `use_pydantic_alias` parameter in pydantic model conversion.
Decides if the all the GraphQL field names for the generated type should use the alias name or not.
```

```python
from pydantic import BaseModel, Field
import strawberry

class UserModel(BaseModel):
      id: int = Field(..., alias="my_alias_name")

@strawberry.experimental.pydantic.type(
    UserModel, use_pydantic_alias=False
)
class User:
    id: strawberry.auto
```

> If use_pydantic_alias is `False`, the graphql type User will have the field id

Reviewer: ```suggestion
If `use_pydantic_alias` is `False`, the GraphQL type User will use `id` for the name of the `id` field coming from the Pydantic model.
```

```
type User {
      id: Int!
}
```

> instead of the field myAliasName.

Reviewer: ```suggestion
With `use_pydantic_alias` set to `True` (the default behaviour) the GraphQL type user will use `myAliasName` for the `id` field coming from the Pydantic models (since the field has a `alias` specified`)
```

```
type User {
      myAliasName: Int!
}
```

> Currently use_pydantic_alias is set to `True` for backwards compatibility.

Reviewer: ```suggestion
`use_pydantic_alias` is set to `True` for backwards compatibility.
```
